tests/README.md

```diff
@@ -56,10 +56,23 @@ pytest with other tests because they have to be run in a Dockerfile:
 - test_s3_to_glue_lambda.py
 - test_setup_external_storage.py
 
-Example)
+
+#### Running tests for lambda
 Run the following command from the repo root to run tests for the lambda function (in develop).
-You can run this locally or inside the docker image.
 
 ```shell script
 python3 -m pytest tests/test_s3_to_glue_lambda.py -v
 ```
+
+#### Running tests for setup external storage
+Run the following command from the repo root to run the integration test for the setup external storage script to check that the STS
+access has been set for a given synapse folder (in develop).
+
+This test takes in two command line arguments:
+
+- test-synapse-folder-id - synapse id of the folder to check STS access for
```

Comment:
I think this is a good opportunity to use the Bridge Downstream Test project: https://www.synapse.org/#!Synapse:syn26721259/wiki/

This is a permanent Synapse project (i.e., it's not created/destroyed programmatically) that we can use to test Synapse functionality that's too complicated to mock (we also use it to maintain test datasets). Rather than requiring a tester to create their own Synapse project and/or folder before running tests, programmatically create/destroy a test folder in this project as part of the test setup/teardown

Reply:
I was hoping this test could be used in more of an integration test way and be used to continuously test (if not continuously test, then check) that there is STS access to the expected synapse projects for recover data each time we deploy or something. Thoughts on that?

Reply:
Sounds good! Made the changes

Reply:
Thanks! But won't you need to do setup/teardown of the test folder? If you use a persistent folder then you may successfully configure the storage location during one test run and not notice breaking changes during the next test run. And the folder you used (syn27558289) is already being used for something else. We don't want to be modifying its external storage location.

Reply:
Also I'm just realizing this is a RECOVER script so it's a bit silly to suggest we use the Bridge Downstream test project for it 😕

Reply:
Hmmm, but we're not configuring the storage location on the test folder?

I was thinking of this test script used in a way like every time we deploy, there's a GH action that runs this test on the synapse folders that are part of the dev pipeline to check that they have the expected STS access.

Similarly, same thing can happen with the staging pipeline in prod.

Because we run `setup_external_storage.py` manually whenever we need to setup STS access so this part where you mention 

> If you use a persistent folder then you may successfully configure the storage location during one test run and not notice breaking changes during the next test run. And the folder you used (syn27558289) is already being used for something else. We don't want to be modifying its external storage location.

won't be a concern?

Reply:
Ohh I see. I have been thinking of everything under `tests/` as quasi-unit tests ("quasi" because sometimes we don't mock things and that puts some of these tests in integration test territory). But although this file is called `test_setup_external_storage`, it's not actually testing our `setup_external_storage` script.

> I was thinking of this test script used in a way like every time we deploy, there's a GH action that runs this test on the synapse folders that are part of the dev pipeline to check that they have the expected STS access.

This is more of an integration test. It's fine to leave it as it is if that's your intention.

Reply:
Yup, I'm not sure if integration tests handled by pytest also belong also under `tests/` (I've seen that being the case before). 

I will leave it as is (will remove the default synapse folder though)